src/main/thrift/parquet.thrift

```diff
@@ -867,12 +867,21 @@ struct ColumnChunk {
     **/
   1: optional string file_path
 
-  /** Byte offset in file_path to the ColumnMetaData **/
-  2: required i64 file_offset
+  /** Deprecated: Byte offset in file_path to the ColumnMetaData
+   *
+   * Past use of this field has been inconsistent, with some implementations
+   * using it to point to the ColumnMetaData and some using it to point to
+   * the first page in the column chunk. In many cases, the ColumnMetaData at this
+   * location is wrong. This field is now deprecated and should not be used.
+   * Writers should set this field to 0 if no ColumnMetaData has been written outside
+   * the footer.
```

Comment:
Hmmm I don't understand would set a required field to 0 is ok and would it cracks the legacy reader

Reply:
Yes, required fields are allowed to have default values (see [this](https://thrift.apache.org/docs/idl#semantics-of-default-values) for some messy details). I looked at code generated by thrift 0.21.0. Java and c++ set the field to the default value in the default constructor, and write the field regardless of its value. The generated rust code doesn't appear to set the default value, however.

It is conceivable that there is a thrift implementation that would not write the '0', which would then break an old reader that expects the field to be set. Maybe I should revert setting a default. @alkis what do you think?

Reply:
This is required by the thrift spec. If stuff is broken, they can fix this in thrift and recompile.

From https://thrift.apache.org/docs/idl#required:

> Write: Required fields are always written and are expected to be set.
> Read: Required fields are always read and are expected to be contained in the input stream.
> Defaults values: are always written